docs/DeveloperGuide.md

```diff
@@ -335,19 +348,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 * 3a. The given index is invalid.
 
-    * 3a1. AddressBook shows an error message.
+    * 3a1. DonnaFin shows an error message.
 
       Use case resumes at step 2.
 
-*{More to be added}*
+**UC03: Locating a contact by name**
+
+**MSS**
+1. User types the command to find the person in the appropriate format.
+2. DonnaFin displays the contacts that match the keyword inputted.
+
+**Extensions**
+* 1a. The user types the command using the wrong format.
+  * 1a1. DonnaFin shows an error message.\
+         Use Case resumes at step 1.
+* 1b. The keyword does not match any contacts.
+  * 1b1. DonnaFin does not display any contact.\
+         Use Case ends.
+
+**UC04: Editing the details of a contact**
+
+**MSS**
+1. User types the command to edit a contact in the appropriate format.
+2. DonnaFin announces that the contact has been successfully updated and displays the new details.
+
+**Extensions**
+* 1a. The user types the command using the wrong format.
+    * 1a1. DonnaFin shows an error message.\
+      Use Case resumes at step 1.
+* 1b. The given index is invalid.
+    * 1b1. DonnaFin shows an error message.\
+      Use Case resumes at step 1.
+  
 
 ### Non-Functional Requirements
 
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
 2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-*{More to be added}*
+4.  A novice with no coding background should be able to use the address book. 
+5.  The system should respond within 100 milliseconds.
+6.  The size of the application should be no more than 100 MB.
+7.  All data should be stored locally.
+8.  The application should not need a remote server to function.
+9.  The application should not require any installer to start functioning.
+10. The GUI should appear fine for screen resolutions 1920x1080 and higher.
 
```

Comment:
main_stream OS can define? that should be it

Reply:
It's already defined but the view is cutoff in GitHub!